### PR TITLE
docs: update middleware package documentation

### PR DESCRIPTION
**Pull request by sergiught**
## Description

This PR is part of a series of PRs aimed at improving the documentation of our publicly exported packages. 

Once all exported structs, vars, funcs are documented we can proceed with enforcing these good practices through the `golangci` linter configuration by enabling the `revive` and `stylecheck` linters.

## References

- https://github.com/openfga/openfga/pull/1261
- https://github.com/openfga/openfga/pull/1263
- https://github.com/openfga/openfga/pull/1264
- https://github.com/openfga/openfga/pull/1280

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/middleware/http/doc.go [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Package http contains utility functions to modify HTTP requests and responses.
```
Comment (jon-whit) [inline pkg/middleware/http/doc.go:1]:
```suggestion
// Package http contains middleware and utility functions to modify HTTP requests and responses.
```

File: pkg/middleware/http/handler.go [hunk 2]
```diff
@@ -18,23 +17,23 @@ import (
 	"github.com/openfga/openfga/pkg/server/errors"
 )
 
-// XHttpCode is used for overriding the standard HTTP code
+// XHttpCode is used for overriding the standard HTTP code.
```
Comment (jon-whit) [inline pkg/middleware/http/handler.go:20]:
```suggestion
// XHttpCode is used to set the header for the response HTTP code.
```

File: pkg/middleware/logging/logging.go [hunk 3]
```diff
@@ -77,10 +78,10 @@ func (r *reporter) PostCall(err error, _ time.Duration) {
 	r.logger.Info(grpcReqCompleteKey, r.fields...)
 }
 
-// PostMsgSend is called once in unary requests and multiple times in streaming requests.
+// PostMsgSend is invoked once in unary requests and multiple times in streaming requests after message sending.
```
Comment (jon-whit) [inline pkg/middleware/logging/logging.go:81]:
```suggestion
// PostMsgSend is invoked once after a unary response or multiple times in streaming requests after each message has been sent.
```

File: pkg/middleware/requestid/doc.go [hunk 4]
```diff
@@ -0,0 +1,2 @@
+// Package requestid contains middleware to log the request ID.
```
Comment (jon-whit) [inline pkg/middleware/requestid/doc.go:1]:
```suggestion
// Package requestid contains middleware to inject and manage request id context.
```

File: pkg/middleware/requestid/requestid.go [hunk 5]
```diff
@@ -15,10 +14,13 @@ import (
 const (
 	requestIDCtxKey   = "request-id-context-key"
 	requestIDTraceKey = "request_id"
-	RequestIDHeader   = "X-Request-Id"
+
+	// RequestIDHeader represents the HTTP header name used for
```
Comment (jon-whit) [inline pkg/middleware/requestid/requestid.go:18]: This is a bit confusing as we're not passing a unique request idenfitier we're returning one.

I'd suggest

```
// RequestIDHeader defines the HTTP header that is set in each HTTP response for a given request. The value of the header is unique per request.
```

File: pkg/middleware/requestid/requestid.go [hunk 6]
```diff
@@ -46,13 +48,13 @@ func reportable() interceptors.CommonReportableFunc {
 		id, _ := uuid.NewRandom()
 		requestID := id.String()
 
-		// Add the requestID to the context
+		// Add the requestID to the context.
```
Comment (jon-whit) [inline pkg/middleware/requestid/requestid.go:51]: This comment and the ones immediately following go without saying. The comments don't add much value. I'd recommend removing them TBH.

File: pkg/middleware/storeid/doc.go [hunk 7]
```diff
@@ -0,0 +1,2 @@
+// Package storeid contains middleware to log the store ID.
```
Comment (jon-whit) [inline pkg/middleware/storeid/doc.go:1]:
```suggestion
// Package storeid contains middleware to inject and manage the store ID context.
```

**Review by miparnisari — COMMENTED** [review]
Now that you know about the `requestid` middleware, I think now you are ready to tackle https://github.com/openfga/openfga/issues/827 😉

**Review by miparnisari — APPROVED** [review]

**Pull request merged**